pytorch_lightning/loops/fit_loop.py

```diff
@@ -181,8 +180,7 @@ def on_advance_start(self) -> None:
             self.trainer.train_dataloader.load_state_dict(self._dataloader_state_dict)
             self._dataloader_state_dict = {}
 
-        # TODO: specify the possible exception
-        with suppress(Exception):
+        if callable(getattr(self.trainer.train_dataloader.sampler, "set_epoch", None)):
```

Comment:
does this assume `sampler` is set? our current dataloader API allows for train_dataloader to be an iterable: https://github.com/PyTorchLightning/pytorch-lightning/blob/ff7305f74d4e097a45b6a4d8c0fff6d4f5aaa386/pytorch_lightning/plugins/training_type/training_type_plugin.py#L204

fyi @ninginthecloud

Reply:
@ananthsub in PyTorch DataLoaders it's not possible to set no sampler, there will always be a default one depending on the what the value of shuffle is. 

Lightning does not support any iterables, only dataloaders and lists or dicts of dataloaders. The type information is: 

```python
TRAIN_DATALOADERS = Union[
    DataLoader,
    Sequence[DataLoader],
    Sequence[Sequence[DataLoader]],
    Sequence[Dict[str, DataLoader]],
    Dict[str, DataLoader],
    Dict[str, Dict[str, DataLoader]],
    Dict[str, Sequence[DataLoader]],
]
```

Reply:
@ananthsub could you or one of your team @daniellepintz @yifuwang @four4fish  open a feature request with your requirements and expectations for adding support for non-DataLoader iterables?